docs/user_guides/fs/storage_connector/usage.md

```diff
@@ -1,9 +1,138 @@
+    
+1. Reading data into spark dataframe
+2. Creating an external feature group
+3. Writing training data
+
+We look at each case in below sections. 
 
 ## Retrieving a Storage Connector
+We retrieve a storage connector simply by its unique name.
+
+=== "PySpark"
+    ```python
+    import hsfs
+    # Connect to the Hopsworks feature store
+    hsfs_connection = hsfs.connection()
+    # Retrieve the metadata handle
+    feature_store = hsfs_connection.get_feature_store()
+    # Retrieve storage connector
+    connector = feature_store.get_storage_connector('connector_name')
+    ```
+
+=== "Scala"
+    ```scala
+    import com.logicalclocks.hsfs._
+    val connection = HopsworksConnection.builder().build();
+    val featureStore = connection.getFeatureStore();
+    // get directly via connector sub-type class e.g. for GCS type
+    val connector = featureStore.getGcsConnector("gcs_test")
+    ```
 
 ## Reading a Spark Dataframe from a Storage Connector
 
+One of the most common usage of a storage connector is to read data directly into a spark dataframe.
+Its achieved via the `read` API of the connector object, which hides all the complexity of authentication and integration 
+with a data storage source. 
+The `read` API primarily has two parameters for specifying the data source, `path` and `query`, depending on the storage connector type.
+The exact behaviour could change depending on the storage connector type, but broadly they could be classified as below
+### Data lake/object based connectors
+
+For data sources based on object/file storage such as AWS S3, ADLS, GCS, we set the full object path to the `path` argument
+and should pass a spark data format to `data_format` argument.
+
+=== "PySpark"
+    ```python
+    # read data into dataframe using path 
+    df = connector.read(data_format='csv',path='path/to/object')   
+    ```
+
+=== "Scala"
+    ```scala 
+    // read data into dataframe using path     
+    val df = connector.read("", "csv" , new HashMap(),"gs://bucket/test/")  
+    ```
+    
+
+### Data warehouse/SQL based connectors
+
+For data sources based on SQL or data warehouses such as Redshift, Snowflake, BigQuery, JDBC, we set a SQL to the `query` 
+argument. Depending on the connector type, we can set the table path on connector and read the whole table without explicitly 
+passing any SQL to `query` argument, e.g. BigQuery.
+
+=== "PySpark"
+    ```python
+    # read results from a SQL 
+    df = connector.read(query='SQL')    
+    # or directly read a table if set on connector
+    df = connector.read()
+    ```
+
+=== "Scala"
+    ```scala 
+    // read results from a SQL 
+    val df = connector.read("SQL", "csv" , new HashMap(),"")    
+    ```
+
+### Streaming based connector
+
+For reading data streams, the Kafka storage connector supports reading a kafka topic into spark streaming dataframe 
+instead of a static dataframe as in other connector types.
```

Comment:
```suggestion
For reading data streams, the Kafka Storage Connector supports reading a Kafka topic into Spark Structured Streaming Dataframes 
instead of a static Dataframe as in other connector types.
```